src/controller/p2p.rs

```diff
@@ -121,6 +151,41 @@ impl<EventStream: Stream<Item = WorkerEvent>> P2PController<EventStream> {
         info!("Heartbeat processing task finished");
     }
 
+    async fn run_assignments_loop(&self, cancellation_token: CancellationToken, assignment_check_interval: Duration) {
+        let mut timer =
+            tokio::time::interval_at(tokio::time::Instant::now(), assignment_check_interval);
+        timer.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        IntervalStream::new(timer)
+            .take_until(cancellation_token.cancelled_owned())
+            .for_each(|_| async move {
+                tracing::debug!("Checking assignment");
+                let network_state_filename = match self.network {
+                    Network::Tethys => "network-state-tethys.json",
+                    Network::Mainnet => "network-state-mainnet.json",
+                };
+                let network_state_url = format!("https://metadata.sqd-datasets.io/{network_state_filename}");
+                let mut latest_assignment = self.latest_assignment.lock().unwrap();
```

Comment:
If you really need to hold the mutex for such a long time, use `try_lock` to not block any threads accidentally.

Reply:
done